pages/api/search/[type].ts

```diff
@@ -16,5 +18,113 @@ const search = async (req: NextApiRequest, res: NextApiResponse) => {
+    // case 'user':
+    //   return await getUser(req, res);
+
+    case 'behavior':
+      if ('id' in req.query) {
+        return await getBehaviorById(req, res);
+      } else {
+        return await getBehaviors(req, res);
+      }
+
+    // has not been tested yet
+    // case 'program':
+    //   if ('id' in req.query) {
+    //     return await getPrograms(req, res);
+    //   } else {
+    //     return await getProgramById(req, res);
+    // }
+
+    // has not been tested yet
+    // case 'report':
+    //   return getReports(req, res);
+
+  }
   res.json([])
-}
\ No newline at end of file
+}
+
+const getBehaviors = async (req: NextApiRequest, res: NextApiResponse) => {
+  try {
+    const behaviors = await prisma.behavior.findMany();
+    return res.status(200).json(behaviors); 
+  } catch (error) {
+    return res.status(500).json({ error });
+  }
+}
+
+const getBehaviorById = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { id } = req.query;
+  try {
+    const behavior = await prisma.behavior.findUnique({
+      where: { id: Number(id) }
+    });
+    return res.status(200).json(behavior);
+  } catch (error) {
+    return res.status(500).json({ error });
+  }
+}
+
+
+// const getPrograms = async (req: NextApiRequest, res: NextApiResponse) => {
+//   try {
+//     const programs = await prisma.program.findMany();
+//     return res.status(200).json(programs);
+//   } catch (error) {
+//     return res.status(500).json({error});
+//   }
+// }
+
+// const getProgramById = async (req: NextApiRequest, res: NextApiResponse) => {
+//   const { id } = req.query;
+//   try {
+//     const program = await prisma.program.findUnique({
+//       where: { id: Number(id) }
+//     });
+//     return res.status(200).json(program);
+//   } catch (error) {
+//     return res.status(500).json({error}); 
+//   }
+// }
+
+// const getReports = async (req: NextApiRequest, res: NextApiResponse) => {
+//   try {
+//     if (req.query.id) {
+//       const report = await prisma.report.findUnique({
+//         where: { id: req.query.id as string },
+//       });
+//       return res.status(200).json(report);
+//     } else {
+//       const reports = await prisma.report.findMany();
+//       return res.status(200).json(reports);
+//     }
+//   } catch (error) {
+//     return res.status(500).json({ error });
+//   }
+// };
+
+// const getUser = async (req: NextApiRequest, res: NextApiResponse) => {
+//   const { id } = req.query;
+//   try {
+//     const user = await prisma.user.findFirst({
+//       where: { id: Number(id) },
+//       data: {
+//         include: {
+          // include employee/patient info
+//           EmployeeProfile: {
+//             include: {
+//               Patients: true
+//             }
+//           },
+//           StudentProfile: true,
+//         }
+//       }
+//     });
+//     return res.status(200).json(user);
+
+//   } catch (error) {
+//     return res.status(500).json({error});
+//   }
+// }
```

Comment:
I would go ahead and remove these comments. If we want to get the old version of the code back we can always look at the commit history.